src/types/cursor_shape_v1.zig

```diff
@@ -0,0 +1,38 @@
+const wlr = @import("../wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+const wp = wayland.server.wp;
+
+pub const CursorShapeManagerV1 = extern struct {
+    global: *wl.Global,
+
+    events: extern struct {
+        request_set_shape: wl.Signal(*RequestSetShapeEvent),
+        destroy: wl.Signal(void),
+    },
+
+    data: ?*anyopaque,
```

Comment:
For consistency with the rest of zig-wlroots this should use a usize for now.
```suggestion
    data: usize,
```
We should really change all of these to `?*anyopaque` at some point though. I think I just used `usize` originally because `@ptrCast()`/`@alignCast()` ergonomics were bad. They've been improved recently though :)